modify-publication/src/main/java/no/unit/nva/publication/ModifyPublicationHandler.java

```diff
@@ -117,18 +112,25 @@ public void handleRequest(InputStream input, OutputStream output, Context contex
                 ContextUtil.injectContext(publicationResponse, publicationContext);
             });
             objectMapper.writeValue(output, new GatewayResponse<>(
-                    objectMapper.writeValueAsString(publicationResponse), headers(), SC_OK));
+                    objectMapper.writeValueAsString(publicationResponse), headers(), OK.getStatusCode()));
         } catch (IOException e) {
             logError(e);
-            objectMapper.writeValue(output, new GatewayResponse<>(objectMapper.writeValueAsString(
-                    Problem.valueOf(BAD_GATEWAY, e.getMessage())), headers(), SC_BAD_GATEWAY));
+            writeErrorResponse(output, BAD_GATEWAY, e);
         } catch (Exception e) {
             logError(e);
-            objectMapper.writeValue(output, new GatewayResponse<>(objectMapper.writeValueAsString(
-                    Problem.valueOf(INTERNAL_SERVER_ERROR, e.getMessage())), headers(), SC_INTERNAL_SERVER_ERROR));
+            writeErrorResponse(output, INTERNAL_SERVER_ERROR, e);
         }
     }
 
+    private void writeErrorResponse(OutputStream output, Status status, String message) throws IOException {
```

Comment:
I was thinking of a library, but we can discuss it later :)

Reply:
Yes, it was first step of creating a common super class for publication handlers